add info about pl/python multidim array support

lisakowen opened the pull request:
added info about pl/python multi-dimensional array support and composite types to pl/python page. not sure if this is too detailed.

lisakowen commented:
@davecramer - would you have time to review these changes?

mkiyama commented:
I think Dave's last day was sometime last week.

hlinnaka commented:
This is essentially identical to the upstream docs (https://www.postgresql.org/docs/devel/static/plpython-data.html#plpython-arrays), just worded slightly differently, right? Did you consider just copy-pasting the upstream docs? That way, it'd be more obvious that it works the same, and we could more easily copy-paste any further enhancements to the upstream docs, too.

lisakowen commented:
@mkiyama - did not know that, thanks for the info.
@danielgustafsson - thanks for reviewing.
@hlinnaka - thanks for reviewing as well.  i created the info from scratch from various sources - the PR, comments, tests, etc.  i missed that section in the docs as i was only searching postgres 8.3 and 8.4.  i will rework the section to better align with the upstream info.

dyozie commented on gpdb-doc/dita/ref_guide/extensions/pl_python.xml:
```diff
@@ -518,6 +524,73 @@ $$ language plpythonu;</codeblock>
+
+<p>PL/Python treats multi-dimensional arrays as lists of lists. You pass a multi-dimensional array to a PL/Python function using nested Python lists. When a PL/Python function returns a multi-dimensional array, the inner lists at each level must all be of the same size. </p>
+
+<p>The following example creates a PL/Python function that takes as input, displays the type of, and returns a multi-dimensional array of integers:</p>
```
This sentence is a little confusing to me.  I'll suggest an edit, but only if it correctly describes things:  "The following example creates a PL/Python function that takes a multi-dimensional array of integers as input. The function displays the type of the provided argument and returns the argument as a multi-dimensional array:"  

Kind of strange, because isn't the type always going to be int?

dyozie commented on gpdb-doc/dita/ref_guide/extensions/pl_python.xml:
```diff
@@ -518,6 +524,73 @@ $$ language plpythonu;</codeblock>
+ {{1,2,3},{4,5,6}}
+(1 row) </codeblock>
+
+<p>PL/Python also accepts other Python sequences, such as tuples, as function arguments for backwards compatibility with Greenplum versions where multi-dimensional arrays were not supported. In such cases, these Python sequences are always treated as one-dimensional arrays because they are ambiguous with composite types.</p>
```
change "these Python sequences" -> "the Python sequences"

lisakowen commented:
@dyozie - i moved arrays, lists and composite types under developing functions along with moving around of some other info.  also incorporated your requested edits.  did not move installing python modules before developing functions - that section is pretty complicated and not necessary for basic function development.  and it could use a bit of work.  let me know what you think. 

(in general, i think this topic could use a restructuring.)
